common/djangoapps/student/tasks.py

```diff
@@ -0,0 +1,125 @@
+"""
+Celery task for course enrollment email
+"""
+import logging
+
+from celery import shared_task
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.course_modes.models import CourseMode
+from common.djangoapps.student.helpers import get_course_dates_for_email, get_instructors
+from lms.djangoapps.utils import get_braze_client
+from openedx.core.djangoapps.catalog.utils import (
+    get_course_uuid_for_course,
+    get_owners_for_course,
+    get_course_run_details
+)
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.features.course_experience import ENABLE_COURSE_GOALS
+
+User = get_user_model()
+log = logging.getLogger(__name__)
+
+MAX_RETRIES = 1
+COUNTDOWN = 5
+
+
+def _remove_date_key_from_course_dates(course_date):
+    _ = course_date.pop('date')
+    return course_date
+
+
+@shared_task(bind=True, ignore_result=True)
+def send_course_enrollment_email(self, user_id, course_id, course_title, short_description, pacing_type):
+    """
+        Send course enrollment email using Braze API
+    """
+
+    course_run_fields = [
+        'key', 'title', 'short_description', 'marketing_url', 'pacing_type', 'min_effort',
+        'max_effort', 'weeks_to_complete', 'enrollment_count', 'image', 'staff',
+    ]
+
+    course_dates_list, course_date_blocks, owners, course_run = [], [], [], None
+
+    try:
+        course_key = CourseKey.from_string(course_id)
+    except InvalidKeyError:
+        course_key = None
+
+    price = CourseMode.min_course_price_for_currency(course_id=course_id, currency='USD')
+    canvas_entry_properties = {
+        'instructors': [],
+        'course_date_blocks': [],
+        'learners_count': '',
+        'partner_image_url': '',
+        'banner_image_url': '',
+        'course_title': course_title,
+        'short_description': short_description,
+        'pacing_type': pacing_type,
+        'goals_enabled': ENABLE_COURSE_GOALS.is_enabled(course_key),
+        'course_run_key': course_id,
+        'course_price': price,
+        'lms_base_url': configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL),
+        'learning_base_url': configuration_helpers.get_value('LEARNING_MICROFRONTEND_URL',
+                                                             settings.LEARNING_MICROFRONTEND_URL)
+    }
+
+    try:
+        user = User.objects.get(id=user_id)
+        course_dates_list = get_course_dates_for_email(user, course_key, request=None)
+        course_date_blocks = list(map(_remove_date_key_from_course_dates, course_dates_list))
```

Comment:
can we do this key removal inside `get_course_dates_for_email` function just to put all dates preparation in the function?